ibc/lightclients/groth16/client_state.go

```diff
@@ -250,38 +250,44 @@ func (cs *ClientState) UpdateState(ctx sdktypes.Context, cdc codec.BinaryCodec,
 		panic(fmt.Sprintf("failed to retrieve trusted consensus state: %s", err))
 	}
 
-	vk, err := DeserializeVerifyingKey(cs.StateTransitionVerifierKey)
-	if err != nil {
-		return []exported.Height{}
-	}
+	// Check if this is a mock proof (all zeros)
+	isMockProof := bytes.Count(header.StateTransitionProof, []byte{0}) == len(header.StateTransitionProof)
```

Comment:
QQ: have you tested that this works? I'm concerned that the entire state transition proof might not be `0` bytes.

One way to test is to generate mock proofs using celestia-prover. Alternatively, you can check out https://github.com/celestiaorg/celestia-zkevm-ibc-demo/pull/56 locally, put it in `mock` mode, generate a proof, and feed it to this function.

Reply:
I didn't get to manually testing this today but I will do it tomorrow morning.

Reply:
manually testing this is blocked on me not being able to install the operator binary.